### PR TITLE
[Dbaas-1060] DBaaS instance & DBaaS Provider metrics

### PR DESCRIPTION
**Pull request by rhkp**
## Description
Rework DBaaSInstance/DBaaSProvider metrics based on the design proposal at this [link](https://docs.google.com/presentation/d/1R9jdTdBwDp70Xlr9r_83nsbG7JUqA8I0DsnKE_N3hsk/edit?usp=sharing)

Screenshots
image

image

image

## Verification Steps(OpenShift Local)

- Install the DBaaS Operator on a fresh OpenShift cluster.
- After successful installation expose the Prometheus service

```
oc project openshift-dbaas-operator
oc expose svc/prometheus-operated
oc get route 
```

- Navigate to Prometheus Service Route in the browser, select a metric such as "dbaas_requests_duration_seconds_sum" and see the resource="dbaas_provider" appear.
- Register MongoDB Provider and create a new MongoDB Instance by following DBaaS Operator User Guide. In the prometheus console, see the resource="dbaas_instance" appear.
- Note: The DBaaSProvider does not have delete handler in the controller, so the delete event is not possible.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by jianrongzhang89** [comment]
@rhkp there are some unit test failures. Could you check?

**Comment by rhkp** [comment]
> @rhkp there are some unit test failures. Could you check?

Hi @jianrongzhang89 thanks, I agree. 

@jeremyary as discussed earlier, please advise, you had some thoughts regarding the same. How do we proceed, so this PR can be approved and merged?

Thanks.

File: controllers/dbaasconnection_controller.go [hunk 1]
```diff
@@ -219,18 +234,27 @@ func mergeConnectionStatus(conn *v1alpha1.DBaaSConnection, providerConn *v1alpha
...
+		metricLabelErrCdValue = metrics.LabelErrorCdValueErrorDeletingConnection
 		return nil
 	}
 	log.Info("connectionObj", "connectionObj", objectKeyFromObject(connectionObj))
 
 	inventory := &v1alpha1.DBaaSInventory{}
 	_ = r.Get(context.TODO(), types.NamespacedName{Namespace: connectionObj.Spec.InventoryRef.Namespace, Name: connectionObj.Spec.InventoryRef.Name}, inventory)
 
-	CleanConnectionMetrics(connectionObj)
+	defer func() {
```
[the remark this answers is not in the document]
Comment (rhkp) [reply controllers/dbaasconnection_controller.go:253]: Okay, sure we can put it there, before line #242.

**Comment by redhatHameed** [comment]
`The DBaaSProvider does not have delete handler in the controller, so the delete event is not possible.`  @rhkp  want to understand why not possible to add delete handler in controller as done for other controllers ?

**Comment by rhkp** [comment]
> `The DBaaSProvider does not have delete handler in the controller, so the delete event is not possible.` @rhkp want to understand why not possible to add delete handler in controller as done for other controllers ?

@redhatHameed Thanks, sure I can look to add this handler.

**Review by jeremyary — APPROVED** [review]

**Pull request merged**